torchdata/datapipes/iter/util/combining.py

```diff
@@ -45,67 +49,159 @@ class IterKeyZipperIterDataPipe(IterDataPipe[T_co]):
         [('a', 101), ('b', 202), ('c', 303)]
     """
 
+    # TODO: remove the default value of key_fns as soon as key_fn and ref_key_fn are removed
     def __init__(
         self,
-        source_datapipe: IterDataPipe,
-        ref_datapipe: IterDataPipe,
-        key_fn: Callable,
-        ref_key_fn: Optional[Callable] = None,
+        *datapipes: IterDataPipe[D],
```

Comment:
Would it be helpful if we change it to `__init__(self, source_datapipe, ref_datapipe, *datapipes, key_fn, ...)` as this datapipe should at least accept 2 datapipe inputs?

Reply:
Somewhat, yes. Although, if possible, I also want to get away from the `ref_datapipe` / `ref_key_fn` names. For me (and other contributors on `torchvision`) the term "reference" implies that this is the main datapipe.